Building on macOS LLVM (clang) 9.1.0 is failing

Didn't look into the problem yet, filing here so it isn't forgotten.

```
$ make
/Applications/Xcode-beta.app/Contents/Developer/usr/bin/make -C ./build V=1 drafter
  c++ '-DBUILDING_DRAFTER=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-DDRAFTER_BUILD_STATIC' -I../ext/boost_1_66_0 -I../ext/sos/src -I../ext/snowcrash/src -I../ext/snowcrash/ext/markdown-parser/src -I../ext/snowcrash/ext/markdown-parser/ext/sundown/src  -Os -gdwarf-2 -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++14 -stdlib=libc++ -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF out/Release/.deps/out/Release/obj.target/libdrafter/src/utils/so/YamlIo.o.d.raw   -c -o out/Release/obj.target/libdrafter/src/utils/so/YamlIo.o ../src/utils/so/YamlIo.cc
In file included from ../src/utils/so/YamlIo.cc:17:
../src/utils/so/../Utf8.h:225:17: error: exception specification of explicitly defaulted default constructor does not match the calculated one
                input_iterator() noexcept = default;
                ^
../src/utils/so/YamlIo.cc:76:23: note: in instantiation of template class 'drafter::utils::utf8::input_iterator<std::__1::__wrap_iter<const char *> >' requested here
        utf8_iterator p{ b, e };
                      ^
../src/utils/so/YamlIo.cc:148:9: note: in instantiation of function template specialization '(anonymous namespace)::escape_yaml_string<std::__1::__wrap_iter<const char *>, std::__1::ostream_iterator<char, char, std::__1::char_traits<char> > >' requested here
        escape_yaml_string( //
        ^
1 error generated.
make[1]: *** [out/Release/obj.target/libdrafter/src/utils/so/YamlIo.o] Error 1
make: *** [drafter] Error 2
```

```
$ clang --version
Apple LLVM version 9.1.0 (clang-902.0.39.1)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

--- a/src/utils/Utf8.h
+++ b/src/utils/Utf8.h
@@ -222,8 +222,6 @@ namespace drafter
                 input_iterator& operator=(const input_iterator& other) = default;
                 input_iterator& operator=(input_iterator&& other) = default;
 
-                input_iterator() noexcept = default;
-
                 friend void swap(input_iterator& lhs, input_iterator& rhs)
                 {
                     using std::swap;

